doc/refman/Classes.tex

```diff
@@ -382,6 +378,54 @@ \subsection{\tt Context {\binder$_1$ \ldots \binder$_n$}}
 Declares variables according to the given binding context, which might
 use implicit generalization (see \ref{SectionContext}).
 
+\asubsection{\tt typeclasses eauto}
+\tacindex{typeclasses eauto}
+
+The {\tt typeclasses eauto} tactic uses a different resolution engine
+than {\tt eauto} and {\tt auto}. The main differences are the following:
+\begin{itemize}
+\item Contrary to {\tt eauto} and {\tt auto}, the resolution is done
+  entirely in the new proof engine (as of Coq v8.6), meaning that
+  backtracking is available among dependent subgoals, and shelving goals
+  is supported.  It analyses the dependencies between subgoals to avoid
+  backtracking on subgoals that are entirely independent.
+\item When called with no arguments, {\tt typeclasses eauto} uses the
+  {\tt typeclass\_instances} database by default (instead of {\tt core})
+  and will try to solve \emph{only} typeclass goals. Other subgoals are
+  automatically shelved and \emph{must be} resolved entirely when the
```

Comment:
Now it does internally, when the typeclass instance generates a subgoal of non-typeclass conclusion. With my upcoming commits it also shelves the initial subgoals that are not typeclasses.